vars/postSuccessfulMetrics.groovy

```diff
@@ -1,5 +1,24 @@
-def call(String metricPrefix) {
-    def long duration = (System.currentTimeMillis() - currentBuild.startTimeInMillis) / 1000;
+/*
+Post multiple metrics
+- metricPrefix defines what you want as the metric identifier
+  e.g. metricPrefix = 'connector.maven-build'
+       ci.ci-4.connector.maven-build.time
+       ci.ci-4.connector.maven-build.success
+- startTime defines start time of item you are timing
+  e.g. startTimeInMillis = 0 
+       This will use current time - current build start time
+  e.g. startTimeInMillis = xxxxx
+       This will use current time - xxxx
+*/
+def call(String metricPrefix, long startTimeInMillis = -1) {
+    def long duration = 0
+
+    if (startTimeInMillis == -1) {
```

Comment:
if the default is meaningless, you may as well just do a null check. In groovy I believe you can just do
```
if (startTimeInMillis) {
 //do the thing
}
```

Reply:
If I don't make a default, the success post doesn't get executed. No groovy error, it just does nothing. I have reinstated the default of -1 and both work again